DSCPullServerAdmin/src/CmdLets/SetDSCPullServerAdminRegistration.cs

```diff
@@ -9,19 +9,22 @@
 
 namespace DSCPullServerAdmin.src.CmdLets
 {
-    [Cmdlet(VerbsCommon.Set,
-        "DSCPullServerAdminRegistration")]
+    [Cmdlet(VerbsCommon.Set, "DSCPullServerAdminRegistration", SupportsShouldProcess = true, ConfirmImpact = ConfirmImpact.High, DefaultParameterSetName = "Id")]
     [OutputType(typeof(void))]
     public class SetDSCPullServerAdminRegistration : BaseCmdlet
     {
-        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true)]
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = "Id")]
         [Alias("Id")]
         [ValidateNotNullOrEmpty()]
-        public string[] AgentId;
+        public string AgentId;
 
         [Parameter(Mandatory = true)]
         [ValidateNotNullOrEmpty()]
-        public string[] Configurations;
+        public string[] ConfigurationNames;
```

Comment:
Sorry to be picky.. Could you make the name singular ConfigurationName. 
Although you can specify multiple values, it is still best practice to have parameter names as singular